packages/firestore/test/unit/local/indexeddb_persistence.test.ts

```diff
@@ -791,6 +791,24 @@ describe('IndexedDbSchema: createOrUpgradeDb', () => {
               );
             });
           })
+          .next(() => {
+            // Verify that we can get recent document changes filtered by read time.
+            const docsRead: string[] = [];
+            const lastReadTime = TEST_SERIALIZER.toDbTimestampKey(version(1));
+            const range = IDBKeyRange.lowerBound(lastReadTime);
+            return remoteDocumentStore
+              .iterate(
+                { index: DbRemoteDocument.readTimeIndex, range },
+                (_, dbRemoteDoc) => {
+                  const doc = TEST_SERIALIZER.fromDbRemoteDocument(dbRemoteDoc);
+                  docsRead.push(doc.key.path.toString());
+                }
+              )
+              .next(() => {
+                // The old documents are not included since readTime was not populated
```

Comment:
Would be nice if this test added some documents with index entries but that were older than version=1 to verify that are not included either...

Reply:
I extracted this into a separate test ("RemoteDocumentCache can filter documents by read time") that does.